packages/commonwealth/client/scripts/views/pages/notification_settings/index.tsx

```diff
@@ -56,6 +60,22 @@ const NotificationSettingsPage = () => {
     relevantSubscribedCommunities,
   } = useNotificationSettings();
 
+  const { mutateAsync: registerClientRegistrationToken } =
+    trpc.subscription.registerClientRegistrationToken.useMutation();
+
+  const handlePushNotificationSubscription = useCallback(() => {
+    async function doAsync() {
+      const permission = await Notification.requestPermission();
+      if (permission === 'granted') {
+        console.log('Notification permission granted.');
+        const token = await getFirebaseMessagingToken();
```

Comment:
What is the purpose of the token here? And why do we get it from firebase? I have a hard time understanding what is going on here in a big picture sense.

Reply:
Ok I think I understand it a bit better now. Knock doesn't actually push notifications to clients. Instead it uses some external service like FCM which does the work instead. If this is the case you can disregard this message, otherwise please correct me.

Reply:
That's correct. This token is needed for FCM